src/index.js

```diff
@@ -520,6 +553,206 @@ class EventsCoreAPI {
    * =========================================================================
    */
 
+  /**
+   * Wrapper to check the event received by webhook
+   * and return decoded (if encoded) and properly parsed payload
+   * @param {*} event event payload received by webhook
+   * @returns decoded and properly parsed payload
+   */
+  __getProperPayload (event) {
+    let decodedJsonPayload
+    try {
+      if (this.__isBase64Encoded(event)) {
+        decodedJsonPayload = JSON.parse(Buffer.from(event, 'base64').toString('utf-8'))
+      } else {
+        // parsing for non-encoded json payloads (e.g. custom events)
+        decodedJsonPayload = JSON.parse(event)
+      }
+    } catch (error) {
+      logger.error('error occured while checking payload' + error.message)
+      return this.__genErrorResponse(400, 'Failed to understand the payload')
+    }
+    return decodedJsonPayload
+  }
+
+  /**
+   * Checks for the payload type if it is base64 encode or not
+   * 
+   * @param {*} eventPayload - the event payload received by the consumer webhook
+   * @returns {boolean} true if encoded or false
+   */
+  __isBase64Encoded (eventPayload) {
+    return Buffer.from(eventPayload, 'base64').toString('base64') === eventPayload
+  }
+
+  /**
+   * Wrapper to fetch the public key (either through aio-lib-state or cloud front url)
+   * and verify the digital signatures
+   * 
+   * @param {*} digiSignature1 - I/O Events generated digital signature1
+   * @param {*} digiSignature2 - I/O Events generated digital signature2
+   * @param {*} pubKeyUrl1 - Cloud Front public key url1
+   * @param {*} pubKeyUrl2 - Cloud Front public key url2
+   * @param {*} recipientClientId - target recipient client id
+   * @param {*} signedPayload - I/O Events proper signed payload
+   * @returns {boolean} true if either signatures are valid or false 
+   */
+  async __verifyDigitalSignature (digiSignature1, digiSignature2, pubKeyUrl1, pubKeyUrl2, recipientClientId, signedPayload) {
+    // fetch the encoded public keys from the shared package default params
+    let pubKey1Pem, pubKey2Pem
+    try {
+      const state = await stateLib.init()
+      pubKey1Pem = await this.__fetchFromCacheOrApi(pubKeyUrl1, state)
+      pubKey2Pem = await this.__fetchFromCacheOrApi(pubKeyUrl2, state)
+    } catch (error) {
+      logger.error('error occurred while fetching from the public key url %s or %s for client %s due to %s',
+        pubKeyUrl1, pubKeyUrl2, recipientClientId, error.message)
+      return this.__genErrorResponse(500, 'Error occurred while fetching Public Key')
```

Comment:
Any reason for returning the same log string in the error response?

Reply:
no reason, looked meaningful for returning to user. may be we can use slightly different error logging messages.